### PR TITLE
fix(richdocuments): Mark some parameters of richdocuments WopiMapper as sensitive

### PR DESCRIPTION
**Pull request by nickvergessen**
Found in our own logs

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: lib/private/Log/ExceptionSerializer.php [hunk 1]
```diff
@@ -39,6 +39,7 @@
 use OCA\Encryption\Hooks\UserHooks;
 use OCA\Encryption\KeyManager;
 use OCA\Encryption\Session;
+use OCA\Richdocuments\Db\WopiMapper;
```
Comment (tcitworld) [inline lib/private/Log/ExceptionSerializer.php:42]: **question:** Won't this explode if richdocuments isn't actived?

Comment (juliushaertl) [reply lib/private/Log/ExceptionSerializer.php:42]: No just if you try to instantiate the class :)

**Review by juliusknorr — APPROVED** [review]

**Comment by rullzer** [comment]
I do get the quick fix. However for future proofing this (and quicker updates). It would be great if there is a way for apps to register their own sensitive parameters. That way it would scale better and you don't have to wait for a server update each time.

**Comment by blizzz** [comment]
> I do get the quick fix. However for future proofing this (and quicker updates). It would be great if there is a way for apps to register their own sensitive parameters. That way it would scale better and you don't have to wait for a server update each time.

We have that: https://github.com/nextcloud/server/blob/master/lib/public/AppFramework/Bootstrap/IRegistrationContext.php#L352

**Comment by nickvergessen** [comment]
> I do get the quick fix. However for future proofing this (and quicker updates). It would be great if there is a way for apps to register their own sensitive parameters. That way it would scale better and you don't have to wait for a server update each time.

Well once PHP 8.2 is the minimum, the correct way is using the native PHP attribute (which I added in https://github.com/nextcloud/richdocuments/pull/3042 )

I was also not aware of the IRegisterContext function, so yeah will use that one instead.

**Pull request closed without merging**